src/js/components/productSupplier/PreferenceTypeColumn.jsx

```diff
@@ -16,20 +18,49 @@ const getLabel = (productSupplierPreferences) => {
     return {
       id: 'react.productSupplier.preferenceType.multiple.label',
       defaultMessage: 'Multiple',
+      icon: <RiInformationLine />,
+      className: 'cell-content',
     };
   }
   return productSupplierPreferences[0].preferenceType?.id;
 };
 
-const PreferenceTypeColumn = ({ productSupplierPreferences }) => {
+const PreferenceTypeColumn = ({ productSupplierPreferences, productSupplierId }) => {
+  const [preferenceTypeModalData, setPreferenceTypeModalData] = useState([]);
   const label = getLabel(productSupplierPreferences);
 
+  const onCellClick = () => {
+    if (productSupplierPreferences.length > 1) {
+      setPreferenceTypeModalData(productSupplierPreferences);
+    }
+  };
+
+  const closeModal = () => setPreferenceTypeModalData([]);
+
   return (
-    <span>
-      {_.isObject(label)
-        ? <Translate id={label.id} defaultMessage={label.defaultMessage} />
-        : label}
-    </span>
+    <>
+      <span
+        className={label?.className}
+        onClick={onCellClick}
+        role="presentation"
+      >
+        {_.isObject(label)
+          ? (
+            <>
+              <Translate id={label.id} defaultMessage={label.defaultMessage} />
+              {' '}
+              {label?.icon}
+            </>
+          )
+          : label}
+      </span>
+      <PreferenceTypeModal
+        productSupplierId={productSupplierId}
+        isOpen={!!preferenceTypeModalData.length}
```

Comment:
I don't mind this, but in general the double negations are difficult to read - how about doing it as:
```js
isOpen={Boolean(preferenceTypeModalData.length)}
```

Reply:
I don't have any preference about that, so I can change it